src/WebJobs.Extensions.DurableTask/EventGridLifeCycleNotificationHelper.cs

```diff
@@ -139,11 +189,33 @@ public HttpMessageHandler HttpMessageHandler
             {
                 httpClient?.Dispose();
                 httpMessageHandler = value;
-                httpClient = new HttpClient(httpMessageHandler);
-                httpClient.DefaultRequestHeaders.Add("aeg-sas-key", this.eventGridKeyValue);
+                httpClient = new HttpClient(httpMessageHandler);
             }
         }
 
+        internal async Task SetUpAuthenticationAsync()
+        {
+            if (this.UseManagedIdentity)
+            {
+                // Use Bearer token for Managed Identity
+                await this.RefreshAccessTokenAsync();

```

Comment:
As far as I can tell, this is the only place where we refresh the access token, and it only happens once when the worker starts up. Suppose the token expires in 1 hour. How do we refresh the token to avoid this expiration?

Reply:
I added another call to `RefreshAccessTokenAsync()` before the POST request in `SendNotificationAsync()`. I checked the [DefaultAzureCredential Class (Azure.Identity)](https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet) docs and it says that `GetTokenAsync()` supports caching and refreshing so I didn't add any other code to handle that. Let me know if you recommend handling caching and refreshing another way.